src/components/Tree/Tree.tsx

```diff
@@ -101,7 +101,12 @@ const reducer = produce((draft: TreeState, action: TreeStateAction) => {
 
         case 'SET_PROJECTION':
             {
+                if (isEqual(draft.projection, action.payload)) {
+                    return;
+                }
+
                 draft.projection = action.payload;
+                return;
```

Comment:
Do we need the added returns in the end? Wouldn't the `break` be enough for the `switch`?

Reply:
no we don't need the `return` it's just a formality, _the immer way_ (https://immerjs.github.io/immer/return)